### PR TITLE
refactor!: use `cfg(nightly)` instead of feature, remove `-Z` flag, use `Diagnostic::try_emit`

### PR DESCRIPTION
**Pull request by MingweiSamuel**
Previous PR (#1587) website build did not work because `panic = "abort"` is set on wasm, leading to aborts for `proc_macro2::Span::unwrap()` calls.

All tests except trybuild seem to pass on stable, WIP #1587 next

BREAKING CHANGE: replaces `hydroflow_lang::diagnostic::Diagnostic::emit` with `try_emit`
BREAKING CHANGE: removes features: `hydroflow/nightly`, `hydroflow_datalog/diagnostics`, `hydroflow_datalog_core/diagnostics`, `hydroflow_lang/diagnostics`, `hydroflow_macro/diagnostics`, `hydroflow_plus/diagnostics`.

### CODE REVIEW

File: hydroflow_lang/src/diagnostic.rs [hunk 1]
```diff
@@ -68,23 +68,47 @@ impl Diagnostic {
...
+        #[cfg(all(nightly, panic = "unwind"))]
         {
-            let pm_diag = match self.level {
-                Level::Error => self.span.unwrap().error(&*self.message),
-                Level::Warning => self.span.unwrap().warning(&*self.message),
-                Level::Note => self.span.unwrap().note(&*self.message),
-                Level::Help => self.span.unwrap().help(&*self.message),
-            };
-            pm_diag.emit();
+            if let Ok(()) = std::panic::catch_unwind(|| {
```
Comment (shadaj) [inline hydroflow_lang/src/diagnostic.rs:76]: :( but okay

**Review by shadaj — APPROVED** [review]

File: hydroflow_lang/src/diagnostic.rs (hunk 1 above)
Comment (MingweiSamuel) [reply hydroflow_lang/src/diagnostic.rs:76]: I could keep the regular emit too? (?)

File: hydroflow_lang/src/diagnostic.rs [hunk 2]
```diff
@@ -68,23 +70,43 @@ impl Diagnostic {
...
+        #[cfg(nightly)]
+        if proc_macro::is_available() {
             let pm_diag = match self.level {
                 Level::Error => self.span.unwrap().error(&*self.message),
                 Level::Warning => self.span.unwrap().warning(&*self.message),
                 Level::Note => self.span.unwrap().note(&*self.message),
                 Level::Help => self.span.unwrap().help(&*self.message),
             };
-            pm_diag.emit();
+            pm_diag.emit()
```
Comment (MingweiSamuel) [inline hydroflow_lang/src/diagnostic.rs:84]: `return Ok(())` disappeared

File: hydroflow_lang/src/pretty_span.rs [hunk 3]
```diff
@@ -1,27 +1,28 @@
...
-                write!(
-                    f,
-                    "{}:{}:{}",
-                    span.source_file().path().display(),
-                    span.start().line(),
-                    span.start().column(),
-                )?;
-                return Ok(());
-            }
+        #[cfg(all(nightly, panic = "unwind"))]
```
Comment (shadaj) [inline hydroflow_lang/src/pretty_span.rs:10]: No longer need `panic = "unwind"`?

Comment (MingweiSamuel) [reply hydroflow_lang/src/pretty_span.rs:10]: Good catch

**Pull request merged**